bridges/snowbridge/primitives/core/src/inbound.rs

```diff
@@ -36,6 +43,8 @@ pub struct Message {
 	pub event_log: Log,
 	/// Inclusion proof for a transaction receipt containing the event log
 	pub proof: Proof,
+	/// Execution update for the block
+	pub update: ExecutionHeaderUpdate,
```

Comment:
The `ExecutionHeaderUpdate` type doesn't make sense in this scenario, as it has fields which are no longer relevant such as `beacon_header` and `ancestry` proof.

Instead:

```suggestion
	pub execution_header:  VersionedExecutionPayloadHeader,
	pub execution_header_branch: Vec<H256>,
```

Reply:
I think we do need these two fields for `verify_ancestry_proof` against a finalized checkpoint on chain.

Reply:
Oops, that's correct. I didn't read the code thoroughly.

We should still reorganise the data structures a bit, for example as shown below. The fields of the `ExecutionHeaderUpdate` struct should be merged into the `Proof` struct, and rename `Proof.data` to `Proof.receipt_proof`.

```
--- a/bridges/snowbridge/primitives/core/src/inbound.rs
+++ b/bridges/snowbridge/primitives/core/src/inbound.rs
@@ -65,10 +65,12 @@ impl Log {
 /// Inclusion proof for a transaction receipt
 #[derive(Clone, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
 pub struct Proof {
-       // The block hash of the block in which the receipt was included.
-       pub block_hash: H256,
+       pub beacon_header: ...
+       pub ancestry_proof: ...
+       pub execution_header: ...
+       pub execution_branch: ...
        // The index of the transaction (and receipt) within the block.
        pub tx_index: u32,
        // Proof keys and values (receipts tree)
-       pub data: (Vec<Vec<u8>>, Vec<Vec<u8>>),
+       pub receipt_proof: (Vec<Vec<u8>>, Vec<Vec<u8>>),
 }
```

Reply:
https://github.com/Snowfork/polkadot-sdk/pull/125/commits/090ec70bb9307302b9d1d09717fa79d8a0db600f